src/hashi/HashiProverUpgradeable.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {HashiProverLib} from "./HashiProverLib.sol";
+import {AccountAndStorageProof, ReceiptProof} from "./HashiProverStructs.sol";
+
+contract HashiProverUpgradeable is Initializable, OwnableUpgradeable {
+    /// @notice Stores the address of the ShoyuBashi contract.
+    /// @dev This address can be updated by the owner using the `setShoyuBashi` function.
+    address public shoyuBashi;
+
+    /**
+     * @notice Emitted when the ShoyuBashi contract address is updated.
+     * @param shoyuBashi The new address of the ShoyuBashi contract.
+     */
+    event ShoyuBashiSet(address shoyuBashi);
+
+    function __HashiProverUpgradeable_init(address shoyuBashi_) public onlyInitializing {
```

Comment:
I believe you would define an `init_unchained` method to allow for inheritance. Not 100% necessary though